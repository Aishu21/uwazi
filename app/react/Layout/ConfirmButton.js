import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Confirm from '../App/Confirm';

class ConfirmButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false
    };
    this.openModal = this.openModal.bind(this);
  }

  openModal() {
    this.setState({ showModal: true });
  }

  render() {
    return (
      <React.Fragment>
        <button onClick={this.openModal}>{this.props.children}</button>
        {this.state.showModal && <Confirm accept={() => this.setState({ showModal: false })}/>}
      </React.Fragment>
    );
  }
}

ConfirmButton.defaultProps = {
  children: ''
};

ConfirmButton.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ])
};

export default ConfirmButton;
