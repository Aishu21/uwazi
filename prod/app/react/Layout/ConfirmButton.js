"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports.default = void 0;var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));

var _ConfirmModal = _interopRequireDefault(require("./ConfirmModal"));function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) {var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {};if (desc.get || desc.set) {Object.defineProperty(newObj, key, desc);} else {newObj[key] = obj[key];}}}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var REACT_ELEMENT_TYPE;function _jsx(type, props, key, children) {if (!REACT_ELEMENT_TYPE) {REACT_ELEMENT_TYPE = typeof Symbol === "function" && Symbol["for"] && Symbol["for"]("react.element") || 0xeac7;}var defaultProps = type && type.defaultProps;var childrenLength = arguments.length - 3;if (!props && childrenLength !== 0) {props = { children: void 0 };}if (props && defaultProps) {for (var propName in defaultProps) {if (props[propName] === void 0) {props[propName] = defaultProps[propName];}}} else if (!props) {props = defaultProps || {};}if (childrenLength === 1) {props.children = children;} else if (childrenLength > 1) {var childArray = new Array(childrenLength);for (var i = 0; i < childrenLength; i++) {childArray[i] = arguments[i + 3];}props.children = childArray;}return { $$typeof: REACT_ELEMENT_TYPE, type: type, key: key === undefined ? null : '' + key, ref: null, props: props, _owner: null };}

class ConfirmButton extends _react.Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false };

    this.openModal = this.openModal.bind(this);
    this.onAccept = this.onAccept.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  onAccept() {
    this.closeModal();
    this.props.action();
  }

  closeModal() {
    this.setState({ showModal: false });
  }

  openModal() {
    this.setState({ showModal: true });
  }

  render() {
    return (
      _jsx(_react.default.Fragment, {}, void 0,
      _jsx("button", { onClick: this.openModal }, void 0, this.props.children),

      this.state.showModal &&
      _jsx(_ConfirmModal.default, {
        message: this.props.message,
        title: this.props.title,
        onAccept: this.onAccept,
        onCancel: this.closeModal })));




  }}


ConfirmButton.defaultProps = {
  children: '',
  message: 'Are you sure you want to continue?',
  title: 'Confirm action',
  action: () => false };var _default =













ConfirmButton;exports.default = _default;