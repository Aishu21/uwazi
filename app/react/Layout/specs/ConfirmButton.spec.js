import React from 'react';

import { shallow } from 'enzyme';

import ConfirmButton from '../ConfirmButton';

describe('ConfirmButton', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {};
  });

  const render = () => {
    component = shallow(<ConfirmButton {...props}>text</ConfirmButton>);
  };

  it('should render a button', () => {
    render();
    expect(component).toMatchSnapshot();
  });

  describe('on click', () => {
    it('should render a Confirm modal', () => {
      component.find('button').simulate('click');
      component.update();
      expect(component).toMatchSnapshot();
    });
  });
});
