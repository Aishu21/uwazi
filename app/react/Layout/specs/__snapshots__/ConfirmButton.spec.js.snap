// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmButton on click should render a Confirm modal 1`] = `
<React.Fragment>
  <button
    onClick={[Function]}
  >
    text
  </button>
  <Confirm
    isOpen={true}
  />
</React.Fragment>
`;

exports[`ConfirmButton should render a button 1`] = `
<React.Fragment>
  <button
    onClick={[Function]}
  >
    text
  </button>
</React.Fragment>
`;
